apps/livreGUI/Controller.cpp

```diff
@@ -149,8 +149,12 @@ struct Controller::Impl
         */
         while( _continuePolling )
         {
-            ScopedLock lock( _subscriberMutex );
-            BOOST_FOREACH( const URISubscriberPair& pair, _subscriberMap )
+            SubscriberMap subscribers;
+            {
+                ScopedLock lock( _subscriberMutex );
+                subscribers = _subscriberMap;
+            }
+            BOOST_FOREACH( const URISubscriberPair& pair, subscribers )
             {
                 zeq::Subscriber* subscriber = pair.second;
                 subscriber->receive( 100 );
```

Comment:
If two threads access the same subscriber for registering events and mean while that subscriber is on receive block this copying may not help.